client/dashboard/data/domain-forwarding.ts

```diff
@@ -0,0 +1,32 @@
+import wp from 'calypso/lib/wp';
+
+export interface DomainForwardingObject {
+	domain_redirect_id: number;
+	domain: string;
+	subdomain: string;
+	fqdn: string;
+	target_host: string;
+	target_path: string;
+	forward_paths: true | false;
+	is_secure: true | false;
+	is_permanent: true | false;
+	is_active?: true | false;
+	source_path?: string;
+}
+
+export interface DomainForwardingSuccessResponse {
+	success: true;
+}
+
+export function fetchDomainForwarding( domainName: string ): Promise< DomainForwardingObject[] > {
+	return wp.req.get( `/sites/all/domain/${ domainName }/redirects?new-endpoint=true` );
+}
+
+export function deleteDomainForwarding(
+	domainName: string,
+	forwardingId: number
+): Promise< DomainForwardingSuccessResponse > {
```

Comment:
Nit: if we never use the return value (because it's always `success: true`?) then maybe we can just simply return `Promise<void>`.

Reply:
Yes, I'll do that